### PR TITLE
WT-6230 Sanitize python test suite directory naming

### PR DESCRIPTION
**Pull request by ddanderson**

### CODE REVIEW

**Comment by ddanderson** [comment]
Before this change, subdirectories of `build_posix/WT_TEST` looked like this.  Notice embedded `(` and `)`.
```
test_config03.test_config03.test_table_ii(1048576.True.None.10.50.False.30.False.None).0
test_config03.test_config03.test_table_ii(1048576.True.None.10.50.False.300.False.None).0
test_config03.test_config03.test_table_ii(1048576.True.None.10.50.True.30.False.None).0
```
Now they look like this:
```
test_config03.test_config03.test_table_ii_1048576.True.None.10.50.False.30.False.None
test_config03.test_config03.test_table_ii_1048576.True.None.10.50.False.300.False.None
test_config03.test_config03.test_table_ii_1048576.True.None.10.50.True.30.False.None
```

File: test/suite/wttest.py [hunk 1]
```diff
@@ -749,6 +760,15 @@ def ttyVerbose(level, message):
...
         return self.id().replace("__main__.","")
 
+    def sanitized_shortid(self):
+        """
+        Return a name that is suitable for creating file system names.
+        In particular, names with scenarios look like
+        'test_file.test_file.test_funcname(scen1.scen2.scen3)'.
+        So transform '(', but remove final ')'.
+        """
+        return self.shortid().translate(str.maketrans('($[]/','_____', ')'))
```
Comment (keitharnoldsmith) [inline test/suite/wttest.py:770]: In other parts of my life, name with spaces often cause problems. Maybe replace that with an underscore, too?

**Review by keitharnoldsmith — APPROVED** [review]
At first I was worried about leaving off the final number (i.e, `.0` in the example from the PR description).  What I didn't realize (maybe it's obvious to other reviewers) is that the test names are unique independent of that.

LGTM!

File: test/suite/wttest.py (hunk 1 above)
Comment (ddanderson) [reply test/suite/wttest.py:770]: Sure, good idea.

**Comment by ddanderson** [comment]
evergreen retry

File: test/suite/wttest.py [hunk 2]
```diff
@@ -749,6 +760,15 @@ def ttyVerbose(level, message):
...
         return self.id().replace("__main__.","")
 
+    def sanitized_shortid(self):
+        """
+        Return a name that is suitable for creating file system names.
+        In particular, names with scenarios look like
+        'test_file.test_file.test_funcname(scen1.scen2.scen3)'.
+        So transform '(', but remove final ')'.
+        """
+        return self.shortid().translate(str.maketrans('($[]/ ','______', ')'))
```
Comment (lukech) [inline test/suite/wttest.py:770]: I'm curious about the reason there are multiple underscore characters in the 2nd str argument.

**Review by lukech — APPROVED** [review]
Thanks for making the change. I only get one non-blocking question.

File: test/suite/wttest.py (hunk 2 above)
Comment (lukech) [reply test/suite/wttest.py:770]: I think I understand -`maketrans()` requires one replacement character to be listed for each character to be replaced in the str argument.

Comment (ddanderson) [reply test/suite/wttest.py:770]: Yes, that's it exactly.

**Pull request merged**